src/Interpreters/BloomFilterHash.h

```diff
@@ -196,18 +196,17 @@ struct BloomFilterHash
             const ColumnString::Chars & data = index_column->getChars();
             const ColumnString::Offsets & offsets = index_column->getOffsets();
 
-            ColumnString::Offset current_offset = pos;
             for (size_t index = 0, size = vec.size(); index < size; ++index)
             {
+                ColumnString::Offset current_offset = offsets[index + pos - 1];
+                size_t length = offsets[index + pos] - current_offset - 1 /* terminating zero */;
                 UInt64 city_hash = CityHash_v1_0_2::CityHash64(
-                    reinterpret_cast<const char *>(&data[current_offset]), offsets[index + pos] - current_offset - 1);
+                    reinterpret_cast<const char *>(&data[current_offset]), length);
```

Comment:
Speaking about this, there are tons of such places, but there is `getDataAt`, and asm looks the same (at least after a quick look), maybe complexity does not worth it?

Reply:
I understand that this may be because getDataAt is a virtual function

Reply:
Good point! But AFAICS the `getDataAt` may be marked `final`, and I guess it is always used after casting to some column type so compiler will optimize this (like here)

Although there can be cases when this will not happen I guess, and in this case explicit writing preferable.

Anyway this is just a thought, not related to this bugfix.